Solink.AddIn.Helpers/RestartableAddInGenerator.cs

```diff
@@ -37,6 +37,7 @@ public RestartableAddInGenerator(string namespaceName, FileInfo sourceAssembly,
 
         public void Generate()
         {
+            AppDomain.CurrentDomain.ReflectionOnlyAssemblyResolve += (sender, args) => Assembly.ReflectionOnlyLoad(args.Name);

```

Comment:
Following just as you suggested created a  nice failing test that passes with the following line,